### PR TITLE
Debugger: enter & exit debug mode and variable explorer to toolbox.

### PR DESCRIPTION
**Pull request by gushuro**
Moved the button to enter/exit debug mode to the simulator toolbar.
Added some options to exit debug mode:
a) Button in the debugger toolbar.
b) Button in top bar (similar to tutorials one).

Moved variable explorer and debugger toolbar to toolbox. They will
replace the usual toolbox when in debug mode.
Removed unnecessary elements in debug mode: Share, help and more in top bar, save, undo, redo in bottom bar.

Debugger v2 0

### CODE REVIEW

**Comment by pelikhan** [comment]
Looks very clean!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: webapp/src/blocks.tsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ export class Editor extends toolboxeditor.ToolboxEditor {
     showSearch: boolean;
     breakpointsByBlock: pxt.Map<number>; // Map block id --> breakpoint ID
     breakpointsSet: number[]; // the IDs of the breakpoints set.
+    debuggerToolboxDiv: any;
```
Comment (pelikhan) [inline webapp/src/blocks.tsx]: Don’t use any

File: webapp/src/blocks.tsx [hunk 2]
```diff
@@ -564,14 +567,28 @@ export class Editor extends toolboxeditor.ToolboxEditor {
...
         if (!immediate) this.toolbox.showLoading();
     }
 
+    updateToolbox(debugging: boolean) {
+        let debuggerToolbox = debugging ? <div>
+            <debug.DebuggerToolbar parent={this.parent} />
+            <debug.DebuggerVariables ref={this.handleDebuggerVariablesRef} parent={this.parent} />
+        </div> : <div />;
+        Util.assert(!!this.debuggerToolboxDiv)
+        debugging ? this.toolbox.hide() : this.toolbox.show();
```
[the remark this answers is not in the document]
Comment (gushuro) [reply webapp/src/blocks.tsx]: Sure

File: webapp/src/container.tsx [hunk 3]
```diff
@@ -375,21 +376,25 @@ export class MainMenu extends data.Component<ISettingsProps, {}> {
         this.props.parent.showReportAbuse();
     }
 
+    toggleDebug() {
+        pxt.tickEvent("simulator.debug", undefined, { interactiveConsent: true });
```
[the remark this answers is not in the document]
Comment (gushuro) [reply webapp/src/container.tsx]: It is. I added a comment to clarify, but the only place where we reference it is as the onClick handler for the `exit debug mode` button in the menu bar.

**Pull request merged**